Bearded.Utilities/Noise/PerlinNoise.cs

```diff
@@ -0,0 +1,98 @@
+using System;
+using System.Linq;
+using Bearded.Utilities.Geometry;
+using OpenTK.Mathematics;
+
+namespace Bearded.Utilities.Noise
+{
+    public static class PerlinNoise
+    {
+        // Use uniformly distributed vectors to force better gradient distribution.
+        private const int numVectorSamples = 12;
+        private static readonly Vector2d[] vectorSamples = vectorSamples = Enumerable.Range(0, numVectorSamples)
+            .Select(i => Direction2.FromRadians(MathConstants.TwoPi * i / numVectorSamples).Vector)
+            .Select(vector => new Vector2d(vector.X, vector.Y))
+            .ToArray();
+
+        public static INoiseMap Generate(int numCellsX, int numCellsY, int? seed)
```

Comment:
It is not clear to me what the numCells parameters do/mean.

Next to that I think this should take a Random, not a seed.

Reply:
I think you need to know something about how Perlin noise works to understand these parameters. I have added a `<summary>` fragment since I think that is the only way to explain how these parameters map to the underlying algorithm.